File: src/net/merayen/elastic/system/intercom/ProcessMessage.java
package net.merayen.elastic.system.intercom;

import java.util.HashMap;
import java.util.Map;

import net.merayen.elastic.util.Postmaster;

/**
 * Message is sent to backend when we want to process a new frame.
 * Backend responds with a new ProcessMessage with the processed data.
 */
public class ProcessMessage extends Postmaster.Message {

	/**
	 * Format: <node-id, <data-identifier, the data>>
	 */
	public final Map<String, Map<String, Object>> data = new HashMap<>();
}

File: src/main/kotlin/net/merayen/elastic/system/intercom/ProcessMessage.kt
package net.merayen.elastic.system.intercom

import java.util.*

/**
 * Message is sent to backend when we want to process a new frame.
 * Backend responds with a new ProcessMessage with the processed data.
 * TODO split this into DoProcessMessage, FinishedProcessMessage
 */
class ProcessMessage : ElasticMessage {
	/**
	 * Data to be sent into the processing backend
	 */
	val input: Map<String, InputFrameData> = HashMap()

	/**
	 * Resulting data from the processing backend
	 */
	var output: Map<String, OutputFrameData> = HashMap()

	/**
	 * Returned by the backend in intervals, to report statistics for UI
	 */
	var statisticsReportMessage: StatisticsReportMessage? = null
}
